Added new cards

EerieCode opened the pull request.

EerieCode committed:
Added new cards

Light Wing Shield
Spiral Flame Strike
Supreme King Gate Zero
Supreme King Gate Infinity
Supreme King Servant Dragon Darkvrm
Supreme King Dragon Zarc

kenpazaraki commented:
Supreme King Gate Zero's scale doesn't seem to work properly. If paired with Infinity it was fine, but if in the other Pendulem Zone there's another monster (in my case Timegazer) its scale becomes 8.

kenpazaraki commented:
Apparently Starving Venom can't be used to summon Zarc using Astrograph/Cronograph's effects.

edo9300 commented:
Zarc's summon procedure is bugged with pendulum+extra deck type monsters. The problem is the get_type function at line 144, because it gets the 1st type of the monster, so not the pendulum. When you have exactly 4 monsters: fusion, synchro, xyz and xyz/pendulum and you try to use the xyz/pendulum before the normal xyz, it will take that card as the xyz material instead of as the pendulum one, making the pendulum material null

EerieCode commented:
That's... going to be complicated. I can't fix it immediately, it will require a lot of testing. Hopefully I'll be able to upload a patch once I return home.

As for your issue, @kenpazaraki , it's probably a duplicate in the database: one has the proper setcode for Starving, the other hasn't, and the result is an error. I'll need to check them and see if that's the case.

EerieCode committed:
Fixed "Supreme King Gate Zero"

LorisC93 commented:
Regarding @edo9300 issue, wouldn't it work if you simply put the pendulum monster choice before the other ones?
Using that order there's no way to create problems, the only thing you'll need to do is to exclude the selected pendulum monster from xyz, synchro and fusion selections.

EerieCode commented:
I don't think it would be enough. Take the case with Starving Venom, Clear Wing, Phantom Dragon and Raging Dragon. Logically you'd want the Pendulum hybrid to be the Xyz, right? But if you select it first, and it's treated as a Pendulum, then the function collapses because it can't find an Xyz.
I'm planning to alter the selection process to force the choice in the correct order (Fusion, Synchro, Xyz, Pendulum): this way, we'll be sure the correct materials are selected. I just need some time to script it and test it.

LorisC93 commented:
ook, I think the main work is to be done with controls, that error can happen with every selection order without preemptive controls.

EerieCode commented:
Yeah, I'm already planning it.

TheOnePharaoh commented:
@EerieCode you need to include  "Miracle Synchro Fusion" in Zarc.
it says here that Miracle Synchro Fusion can be used
http://yugioh.wikia.com/wiki/Card_Tips:Supreme_King_Dragon_Zarc

EerieCode committed:
Fixed "Supreme King Dragon Zarc"

EerieCode committed:
Added recent errata

Cyber Blader
Rescue Cat
Brionac, Dragon of the Ice Barrier
Imperial Order
Goyo Guardian
Future Fusion
Brain Control

EerieCode committed:
Added new cards (credits to dest)

Spiral Hold of the Mythic Radiance Dragon
Clashing Whirlpool of the Mythic Radiance Dragon
Deep Whirlpool of the Mythic Radiance Dragon

EerieCode committed:
Added new cards (credits to mercury233)

Pacifis, City of Mythic Radiance
Spiral Crash of the Mythic Radiance Dragon

EerieCode committed:
Added new cards

Cosmic Flare
Passionate Duelists
Pendulum Fusion

EerieCode commented:
Aaaaaaaand... done! The only one missing is "White Veil", so might as well wait until I add it before approving this!

EerieCode committed:
Added new cards

White Veil
Spiral, the Mythic Radiance Dragon

kenpazaraki commented:
Supreme King Gate Zero's effect that searches for a Poly/Fusion card can't be activated.
Anyway, thank you!

edo9300 commented:
so many new cards! xD anyway eerie i think shaim's isn't a bad idea, you choose the pendulum monster for first, then the others
